pkg/machine/gvproxy_windows.go

```diff
@@ -0,0 +1,42 @@
+package machine
+
+import (
+	"os"
+	"time"
+
+	"github.com/containers/winquit/pkg/winquit"
+	"github.com/sirupsen/logrus"
+)
+
+func waitOnProcess(processID int) error {
+	logrus.Infof("Going to stop gvproxy (PID %d)", processID)
+
+	p, err := os.FindProcess(processID)
+	if err != nil {
+		return nil
+	}
+
+	// Gracefully quit and force kill after 30 seconds
+	if err := winquit.QuitProcess(processID, 30*time.Second); err != nil {
+		return err
+	}
+
+	logrus.Debugf("completed grace quit || kill of gvproxy (PID %d)", processID)
+
+	// Make sure the process is gone
+	done := make(chan struct{})
+	go func() {
+		_, _ = p.Wait()
+		done <- struct{}{}
+	}()
+
+	select {
+	case <-done:
+		logrus.Debugf("verified gvproxy termination (PID %d)", processID)
+	case <-time.After(10 * time.Second):
+		// Very unlikely but track just in case
+		logrus.Errorf("was not able to kill gvproxy (PID %d)", processID)
+	}
```

Comment:
This seem odd, winquit.QuitProcess() seem to already do this internally so why is this function doing the same thing twice with DIFFERENT timeout values!
It really seem like this whole function should juts call winquit.QuitProcess() and not add extra duplicated logic on top.

Reply:
This is intentional as well. QuitProcess provides a softer guarantee than what this function is looking for. winquit.QuitProcess only guarantees the process will eventually die (and soon), whereas this function is intending to guarantee the process is truly gone after returning (and all associated resources have been cleaned up). The rationale for that is to prevent flakes from successive calls to machine stop and start.  While QuitProcess does have a timeout, it's only used to attempt a graceful close (similar to SIGTERM). In the event that the graceful close doesnt happen in the expected time a hard quit is sent and the kernel forces immediate cleanup (similar to SIGKILL). That tends to be near immediate, but can be slightly deferred if the system is under heavy load. This is why the latter timeout is smaller, the former graceful process is expected to take longer since it relies on application execution logic and typically 3 way closes vs RSTs. That said I agree a caller shouldn't have to do this so will add a QuitProcessConfirmExit in the future. In hindsight perhaps that should have been the base logic of QuitProcess, although the typical contract of Kill() on most (all) operating systems is async scheduling of kill not blocking until exit.